packages/flutter_tools/templates/app_shared/windows.tmpl/CMakeLists.txt.tmpl

```diff
@@ -8,7 +8,11 @@ set(BINARY_NAME "{{projectName}}")
 
 # Explicitly opt in to modern CMake behaviors to avoid warnings with recent
 # versions of CMake.
-cmake_policy(SET CMP0063 NEW)
+if(${CMAKE_VERSION} VERSION_LESS "3.24")
+  cmake_policy(SET CMP0063 NEW)
+else()
```

Comment:
This logic looks wrong; why are you removing 63 for new versions? Doesn't that just trade one warning for another one?

Is there a reason not to use the simpler cmake-version-based variant that opts into all new behaviors, as I suggested in the issue?

Reply:
Well, this is odd. Not only does that not work in the top-level project, but setting it directly in the top level project also doesn't work. I have to set it in the plugin's CMakeLists.txt to eliminate the warning locally (`VERSION` also works in that context). I thought the setting would be inherited down the build hierarchy by default, and the new scopes would only allow locally overriding what was inherited.

So either I'm missing something about how CMake works here (which is certainly possible), or it's not actually possible to fix this except at the plugin level. (In which case we should add this to the plugin example template, since we use downloads there.)